src/core/ext/client_channel/parse_address.c

```diff
@@ -49,12 +49,11 @@
 
 int parse_unix(grpc_uri *uri, grpc_resolved_address *resolved_addr) {
   struct sockaddr_un *un = (struct sockaddr_un *)resolved_addr->addr;
-
+  const size_t maxlen = sizeof(un->sun_path);
+  if (strnlen(uri->path, maxlen) == maxlen) return 0;
```

Comment:
If the size is exactly maxlen, we have a problem: the \0 would be one-past the end